framework/include/variables/MooseVariableFV.h

```diff
@@ -250,6 +277,28 @@ class MooseVariableFV : public MooseVariableField<OutputType>
     checkIndexingScalingCompatibility();
     return _element_data->adGradSln();
   }
+
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  /**
+   * Retrieve (or potentially compute) the gradient on the provided element
+   * @param elem The element for which to retrieve the gradient
+   */
+  const VectorValue<ADReal> & adGradSln(const Elem * const elem) const;
+
+  /**
+   * Retrieve (or potentially compute) the gradient on the provided face
+   * @param face The face for which to retrieve the gradient
+   */
+  const VectorValue<ADReal> & adGradSln(const FaceInfo & fi) const;
```

Comment:
This doc comment should explain that you are getting a cross-diffusion-corrected face gradient.